Simplenote/src/main/java/com/automattic/simplenote/viewmodels/AddTagViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.automattic.simplenote.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.automattic.simplenote.R
+import com.automattic.simplenote.models.Tag
+import com.automattic.simplenote.utils.TagUtils
+import com.simperium.client.Bucket
+import com.simperium.client.BucketObjectNameInvalid
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+
+class AddTagViewModel(
+        private val tagsBucket: Bucket<Tag>,
+        defaultDispatcher: CoroutineDispatcher = Dispatchers.Default
+) : ScopedViewModel(defaultDispatcher) {
+
+    private val _showKeyboard = MutableLiveData<Boolean>()
+    val showKeyboard: LiveData<Boolean> = _showKeyboard
+
+    private val _tagError = MutableLiveData<Int?>()
+    val tagError: LiveData<Int?> = _tagError
+
+    private val _isResultOK = MutableLiveData<Boolean>()
+    val isResultOK: LiveData<Boolean> = _isResultOK
+
+    fun validateTag(tagName: String) {
+        if (tagName.isEmpty()) {
+            _tagError.value = R.string.tag_error_empty
+            return
+        }
+
+        if (tagName.contains(" ")) {
+            _tagError.value = R.string.tag_error_spaces
+            return
+        }
+
+        if (!TagUtils.hashTagValid(tagName)) {
+            _tagError.value = R.string.tag_error_length
+            return
+        }
+
+        if (!TagUtils.isTagMissing(tagsBucket, tagName)) {
+            _tagError.value = R.string.tag_error_exists
+            return
+        }
+
+        _tagError.value = null
+    }
+
+    fun saveTag(tagName: String) {
+        try {
+            _showKeyboard.value = false
+
+            TagUtils.createTagIfMissing(tagsBucket, tagName)
```

Comment:
If you implement the unit tests, this would've to be wrapped.

Reply:
Fixed in d3a5980c7eaad1273fe417ffac4786a8c2d56ea0.